src/AudioManagerImpl.js

```diff
@@ -0,0 +1,210 @@
+import type {AnnouncedSoundName, AudioManager} from './types';
+
+type AnnouncementLookupTable = {
+    forward1: Player,
+    forward2: Player,
+    forward3: Player,
+    left45: Player,
+    left90: Player,
+    left180: Player,
+    right45: Player,
+    right90: Player,
+    right180: Player,
+    add: Player,
+    deleteAll: Player,
+    delete: Player,
+    moveToPrevious: Player,
+    moveToNext: Player,
+    replace: Player
+}
+
+
+const AnnouncementDefs = new Map<string, string>([
+    ['forward1', '/audio/Move.wav'],
+    ['forward2', '/audio/Move.wav'],
+    ['forward3', '/audio/Move.wav'],
+    ['left45', '/audio/TurnLeft.wav'],
+    ['left90', '/audio/TurnLeft.wav'],
+    ['left180', '/audio/TurnLeft.wav'],
+    ['right45', '/audio/TurnRight.wav'],
+    ['right90', '/audio/TurnRight.wav'],
+    ['right180', '/audio/TurnRight.wav'],
+    ['add', './audio/AddMovement.wav'],
+    ['deleteAll', '/audio/DeleteAll.wav'],
+    ['delete', '/audio/DeleteMovement.wav'],
+    ['moveToPrevious', '/audio/MoveToLeft.wav'],
+    ['moveToNext', '/audio/MoveToRight.wav'],
+    ['replace', '/audio/ReplaceMovement.wav']
+]);
+
+function octaveModulo (rawPitch: number) : number {
+    const adjustedPitch = rawPitch % 12;
+    return adjustedPitch < 0 ? adjustedPitch + 12 : adjustedPitch;
+}
+
+// Modified "Tonnetz" tuning, see https://en.wikipedia.org/wiki/Tonnetz for explanation and diagrams.
+export function getNoteForState (characterState: CharacterState) : string {
+    // The centre note (xPos: 0, yPos: 0) is 440hz = A4 = 69.
+
+    // Every "column" is 7 tones from the next but stays within the same octave.  This results in a pattern that
+    // cycles through the full octave range every 12 squares.
+    const xPitchOffset = octaveModulo(7 * characterState.xPos);
+
+
+    // Every "row" is 4 tones from the next but stays within the same octave.  This results in a pattern of three
+    // notes that repeats every three rows.
+    const yPitchOffset = octaveModulo(4 * characterState.yPos);
+
+    const combinedPitchOffset = octaveModulo(xPitchOffset + yPitchOffset);
+
+    // To vary the range of notes without going too high or low, we use the repeating nature of the "row" pattern
+    // to divide the tuning into "octave bands" every three rows.  The middle band (yPos of -1, 0, or 1) is octave 3.
+    // The "band" above centre (yPos of 2, 3, or 4) is octave 4.  Anything higher is octave 5. The "band" below
+    // centre (yPos of -2, -3, or -4) is octave 2.  Anything lower is octave 1.
+    const octaveOffset = (Math.round(characterState.yPos / 3));
+    const boundedOctaveOffset = Math.max(Math.min(3, octaveOffset), -3);
+    const octave = 4 - boundedOctaveOffset;
+
+    // const midiNote = (12 * octave);
+    const midiNote = combinedPitchOffset + (12 * octave);
+    const noteName: string = Midi(midiNote).toNote();
+
+    return noteName;
+}
+
+export default class AudioManagerImpl
+  implements AudioManager {
```

Comment:
You've changed a lot of the indentation on this file to 2 spaces. Please use 4 spaces instead.

This is something that ESLint could help us with. I've created a JIRA to add rules to our `eslintConfig` to check indentation: https://issues.fluidproject.org/browse/C2LC-277

I did a little experimenting this morning and the following seems like a good starting place:

    "indent": ["error", 4, {
        "CallExpression": {"arguments": "off"},
        "SwitchCase": 1
    }],

This still results in a number of lines that don't match, so there will be some work to do to align the code with the rules.

Reply:
Yes, I didn't do it on purpose, linting rules are a really good idea.  I'll make a small ticket / pull to clean them up.

Reply:
I'll still add them as part of this pull, actually, but I think a ticket it good to tag the commits with.

Reply:
Just found [C2LC-277](https://issues.fluidproject.org/browse/C2LC-277), I'll pick up that and use it to tag both the added rules and the cleanup..

Reply:
I think I'll make a separate small pull after all, there are a couple of edge cases that I'd like to flag and discuss separately.